File: CS/CustomEditorWinApp3/CustomLookUpEdit.cs
using System;
using System.ComponentModel;
using DevExpress.XtraEditors;

namespace CustomEditor
{
	class CustomLookUpEdit : LookUpEdit
	{
		static CustomLookUpEdit()
		{
			RepositoryItemCustomLookUpEdit.Register();
		}
		public CustomLookUpEdit()
			: base()
		{
		}

		[DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
		public new RepositoryItemCustomLookUpEdit Properties
		{
			get
			{
				return base.Properties as RepositoryItemCustomLookUpEdit;
			}
		}

		public override string EditorTypeName
		{
			get
			{
				return RepositoryItemCustomLookUpEdit.EditorName;
			}
		}

		protected override bool ProcessNewValueCore(bool partital, string text)
		{
			if ( this.Properties.IsNullInputAllowed && String.IsNullOrEmpty(text) )
			{
				this.EditValue = null;
				return true;
			}

			return base.ProcessNewValueCore(partital, text);
		}
	}
}

File: VB/CustomEditorWinApp3/CustomLookUpEdit.vb
Imports System.ComponentModel
Imports DevExpress.XtraEditors

Namespace CustomEditor

    Friend Class CustomLookUpEdit
        Inherits LookUpEdit

        Shared Sub New()
            RepositoryItemCustomLookUpEdit.Register()
        End Sub

        Public Sub New()
            MyBase.New()
        End Sub

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemCustomLookUpEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomLookUpEdit)
            End Get
        End Property

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemCustomLookUpEdit.EditorName
            End Get
        End Property

        Protected Overrides Function ProcessNewValueCore(ByVal partital As Boolean, ByVal text As String) As Boolean
            If Properties.IsNullInputAllowed AndAlso String.IsNullOrEmpty(text) Then
                EditValue = Nothing
                Return True
            End If

            Return MyBase.ProcessNewValueCore(partital, text)
        End Function
    End Class
End Namespace
